herald/base.py

```diff
@@ -154,6 +158,15 @@ def resend(cls, sent_notification, raise_exception=False):
         returns boolean whether or not the notification was sent successfully
         """
 
+        # handle skipping a notification based on user preference
+        if hasattr(sent_notification.user, 'usernotification'):
+            notifications = sent_notification.user.usernotification
```

Comment:
I don't think this is correct. I think if the herald is installed, all users will have a `usernotification` attribute. But `sent_notification.user.usernotification` would raise a `DoesNotExist` if it is not there. I might be wrong though. Probably need a test to check that.

Reply:
I'm wrong about this. It seems to work.